### PR TITLE
Add leader transfer balance strategy.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**

### CODE REVIEW

**Comment by qiuyesuifeng** [comment]
@huachaohuang @shenli @siddontang  PTAL

**Comment by huachaohuang** [comment]
LGTM

File: conf/config.toml [hunk 1]
```diff
@@ -17,10 +17,12 @@ metric-addr = ""
 
 min-capacity-used-ratio = 0.4
 max-capacity-used-ratio = 0.9
+max-leader-count = 10
```
Comment (siddontang) [inline conf/config.toml]: I think 10 is too small.

File: server/balancer.go [hunk 2]
```diff
@@ -340,6 +368,49 @@ func (rb *resourceBalancer) Balance(cluster *clusterInfo) (*balanceOperator, err
...
+	}
+
+	// If region peer count is not equal to max peer count, no need to do leader transfer.
+	if len(region.GetPeers()) != int(cluster.getMeta().GetMaxPeerCount()) {
+		log.Warnf("region peer count %d not equals to max peer count %d, no need to do leader transfer",
+			len(region.GetPeers()), cluster.getMeta().GetMaxPeerCount())
+		return nil, nil
+	}
+
+	if !rb.checkScore(cluster, leader, newLeader, true) {
```
Comment (siddontang) [inline server/balancer.go]: no need to check leader diff here? E.g, transferring from store 1 -> 2, now 1 has 11 leaders, 2 has 10 leaders, but 1 and 2 has different capacity so checkScore returns true.

**Comment by qiuyesuifeng** [comment]
We should discuss a new method to handle balance.
Close it now.

**Pull request closed without merging**